### PR TITLE
feat: Change to Hashdit

### PR DESCRIPTION
**Pull request by ChefMomota**
### Summary
🔗🛡️📊

This pull request replaces AvengerDAO with Hashdit as the risk scan service provider for tokens on the Swap view. It updates the `AccessRisk` component, the `fetchTokenRisk` hook, and the translation keys to use the new API and display more detailed and accurate risk levels. It also deletes the unused `apps/web/src/pages/api/risk/[chainId]/[address].ts` file.

> _We scan the tokens of doom_
> _With Hashdit's blazing tool_
> _We show the risk in the tooltip_
> _And we don't trust Avenger's fool_

### Walkthrough
*  Update the risk scan service provider from AvengerDAO to Hashdit ([link](https://github.com/pancakeswap/pancake-frontend/pull/7385/files?diff=unified&w=0#diff-7238357db46da6bb48ca41841457ac0e832ee21fb8fd76a6b3c9dbf701ce8c62L39-R39), [link](https://github.com/pancakeswap/pancake-frontend/pull/7385/files?diff=unified&w=0#diff-5fd8372e743880064fc3923838aa22af2791a24597a6cc87e59dac0f9b346e25L118-R129), [link](https://github.com/pancakeswap/pancake-frontend/pull/7385/files?diff=unified&w=0#diff-b25d019d7289b09ba9c3c074126d3c93959f079bb13a0655dc9819cbf387974dL1-R34), [link](https://github.com/pancakeswap/pancake-frontend/pull/7385/files?diff=unified&w=0#diff-b25d019d7289b09ba9c3c074126d3c93959f079bb13a0655dc9819cbf387974dL51-R51), [link](https://github.com/pancakeswap/pancake-frontend/pull/7385/files?diff=unified&w=0#diff-eb2ab983e2cefc22516cb7814c86346f4b2bd2b971980952868173095d48f459L2126), [link](https://github.com/pancakeswap/pancake-frontend/pull/7385/files?diff=unified&w=0#diff-eb2ab983e2cefc22516cb7814c86346f4b2bd2b971980952868173095d48f459L2345-L2347), [link](https://github.com/pancakeswap/pancake-frontend/pull/7385/files?diff=unified&w=0#diff-eb2ab983e2cefc22516cb7814c86346f4b2bd2b971980952868173095d48f459L2611-R2610))
*  Add optional chaining for the token parameter in the fetchRiskToken hook to prevent errors ([link](https://github.com/pancakeswap/pancake-frontend/pull/7385/files?diff=unified&w=0#diff-5fd8372e743880064fc3923838aa22af2791a24597a6cc87e59dac0f9b346e25L79-R89))
*  Move the TOKEN_RISK_T constant to the top of the AccessRisk component file for better readability and reusability ([link](https://github.com/pancakeswap/pancake-frontend/pull/7385/files?diff=unified&w=0#diff-5fd8372e743880064fc3923838aa22af2791a24597a6cc87e59dac0f9b346e25R33-R42), [link](https://github.com/pancakeswap/pancake-frontend/pull/7385/files?diff=unified&w=0#diff-5fd8372e743880064fc3923838aa22af2791a24597a6cc87e59dac0f9b346e25L90-L98))
*  Delete the unused api/risk/[chainId]/[address].ts file ([link](https://github.com/pancakeswap/pancake-frontend/pull/7385/files?diff=unified&w=0#diff-ec27de19b451e65fcc87aee4dbe743649a9ab558b5a5cdb0ef92e73381aae06f))

### CODE REVIEW

File: apps/web/src/components/Menu/GlobalSettings/SettingsModal.tsx [hunk 1]
```diff
@@ -46,6 +45,8 @@ import {
 import { AtomBox } from '@pancakeswap/ui'
 import { useMMLinkedPoolByDefault } from 'state/user/mmLinkedPool'
 import styled from 'styled-components'
+import { TOKEN_RISK } from 'views/Swap/hooks/fetchTokenRisk'
```
Comment (chefjackson) [inline apps/web/src/components/Menu/GlobalSettings/SettingsModal.tsx:48]: We better move the risk related constant and component to `src/components` if we want to use them in a common component.

When importing modules, we better keep in mind the dependency relationship between the modules.

Normally we shouldn't rely on a specific business component from a commonly used component. In this case from the import statement, it implies that the Menu component relies on Swap feature, which is not what we want and could cause further issue like circular reference

Comment (ChefMomota) [reply apps/web/src/components/Menu/GlobalSettings/SettingsModal.tsx]: done

File: apps/web/src/views/Swap/components/AccessRisk/index.tsx [hunk 2]
```diff
@@ -7,19 +7,20 @@ import {
...
+  AutoRenewIcon,
 } from '@pancakeswap/uikit'
-import { useMemo, useState } from 'react'
+import { useEffect, useMemo, useState } from 'react'
 import { useUserTokenRisk } from 'state/user/hooks/useUserTokenRisk'
 import { useAllLists } from 'state/lists/hooks'
 import styled from 'styled-components'
 import useSWRImmutable from 'swr/immutable'
-import { fetchRiskToken, TOKEN_RISK } from 'views/Swap/hooks/fetchTokenRisk'
+import { fetchRiskToken } from 'views/Swap/hooks/fetchTokenRisk'
```
Comment (chefjackson) [inline apps/web/src/views/Swap/components/AccessRisk/index.tsx:22]: Is this hook related to swap feature? If not can we treat it as a common hook and move it to `src/hooks`?

File: apps/web/src/views/Swap/components/AccessRisk/index.tsx
Comment (chefjackson) [inline apps/web/src/views/Swap/components/AccessRisk/index.tsx:1]: This whole component looks nothing to do with Swap. Can we move it to `src/components`?

Comment (ChefMomota) [reply apps/web/src/views/Swap/components/AccessRisk/index.tsx:1]: Done both.

**Review by chefjackson — APPROVED** [review]
